zipline/data/loader.py

```diff
@@ -204,6 +204,15 @@ def ensure_benchmark_data(symbol, first_date, last_date, now, trading_day):
     path.
     """
     path = get_data_filepath(get_benchmark_filename(symbol))
+
+    # If the path does not exist, it means the first download has not happened
+    # yet, so trigger a download and write it to 'path'.
+    if not os.path.exists(path):
+        data = get_benchmark_returns(
+            symbol, first_date - trading_day, last_date,
+        )
+        data.to_csv(path)
+
```

Comment:
This is duplicating the work done at the bottom of the function.  I think what you want to do here is
```
if os.path.exists(path):
    <try/except block>
```